src/Moryx.Products.UI.Interaction/Recipes/Details/RecipeDetailsViewModelBase.cs

```diff
@@ -81,16 +81,19 @@ public async Task Load(RecipeViewModel recipeVm, IReadOnlyCollection<WorkplanVie
             await LoadTypeAndWorkplans(recipeVm.Model, workplans);
 
             WorkplanViewModel wp = null;
-            if (recipeVm.Model.WorkplanId != 0)
-                wp = workplans.FirstOrDefault(w => w.Id == recipeVm.Model.WorkplanId);
+            if (recipeVm.Model.WorkplanModel.Id != 0)
```

Comment:
Careful with compatibility. An old server might not yet transmit that value. Use the new one like `Workplan?.Id ?? WorkplanId` for this major.